app/workers/segment_delete_user_worker.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class SegmentDeleteUserWorker < ActiveJob::Base # rubocop:disable Rails/ApplicationJob # That would be Rails 5 :)
+  class MissingUserIDError < StandardError; end
+
+  def perform(event_id)
+    event = EventStore::Repository.find_event!(event_id)
+    raise(MissingUserIDError) unless (user_id = event.data[:user_id])
```

Comment:
No need of this error, the event should be invalid if there is no user_id

Reply:
Yeah, the summary of what we spoke regarding to this class is in https://github.com/3scale/porta/pull/717#discussion_r271664315 😄